lib/mpool/lib/mblock_file.c

```diff
@@ -601,20 +602,30 @@ mblock_wlen_get(const struct mblock_file *mbfp, uint64_t mbid)
 }
 
 static HSE_ALWAYS_INLINE bool
-mblock_is_prealloc(const struct mblock_file *mbfp, uint64_t mbid)
+mblock_is_prealloced(const struct mblock_file *mbfp, uint64_t mbid)
+{
+    const uint32_t wlen = atomic_read(mbfp->wlenv + block_id(mbid));
+
+    return ((wlen & MBLOCK_WLEN_PREALLOC_MASK) >> MBLOCK_WLEN_PREALLOC_SHIFT) == 1;
+}
+
+static HSE_ALWAYS_INLINE bool
+mblock_is_punched(const struct mblock_file *mbfp, uint64_t mbid)
 {
     const uint32_t wlen = atomic_read(mbfp->wlenv + block_id(mbid));
 
-    return (wlen >> MBLOCK_WLEN_PREALLOC_SHIFT) == 1;
+    return ((wlen & MBLOCK_WLEN_PUNCH_MASK) >> MBLOCK_WLEN_PUNCH_SHIFT) == 1;
```

Comment:
It is interesting that we choose to encode flags in the wlen.

Reply:
Yes, our max supported mblock size is 1G but we use a uint32_t to store wlen both in-memory and on-media. These two flags are also closely related to how wlen is computed for each mblock, so made sense to use these two unused bits for this purpose.